File: EstudoTypescript /linked-list.js
var MeuNode = /** @class */ (function () {
    function MeuNode(data, next) {
        if (next === void 0) { next = null; }
        this.data = data;
        this.next = next;
    }
    return MeuNode;
}());
function insertAtFront(head_ref, new_data) {
    var new_node = new MeuNode(new_data);
    new_node.next = head_ref[0];
    head_ref[0] = new_node;
}
function removeFistNode(head) {
    if (head == null)
        return null;
    var temp = head;
    head = head.next;
    return head;
}
function printList(node) {
    var current = node;
    if (node === null) {
        console.log("a lista esta vazia");
        return;
    }
    while (current !== null) {
        console.log(" " + current.next);
        current = current.next;
    }
    console.log("\n");
}
function main() {
    var head = [];
    insertAtFront(head, 1);
    insertAtFront(head, 2);
    insertAtFront(head, 3);
    insertAtFront(head, 3);
    insertAtFront(head, 4);
    insertAtFront(head, 5);
    insertAtFront(head, 6);
    insertAtFront(head, 7);
    console.log("after removing first node");
    printList(head[0]);
    head[0] = removeFistNode(head[0]);
    console.log("depois de remover");
    printList(head[0]);
}
main();

File: EstudoDart/linked-list.dart
class Node<T> {
  T? value;
  Node<T>? next;

  Node(this.value);
}

class LinkedList<T> {
  Node<T>? head;

  LinkedList();

  bool get isEmpty => head == null;

  void add(T value) {
    final newNode = Node<T>(value);
    if (isEmpty) {
      head = newNode;
    } else {
      var current = head;
      while (current!.next != null) {
        current = current.next;
      }
      current.next = newNode;
    }
  }

  void remove(T value) {
    if (isEmpty) return;

    if (head!.value == value) {
      head = head!.next;
      return;
    }
    var current = head;
    while (current!.next != null) {
      if (current.next!.value == value) {
        current.next = current.next!.next;
        return;
      }
      current = current.next;
    }
  }

  void printList() {
    var current = head;
    while (current != null) {
      print(current.value);
      current = current.next;
    }
  }
}
void main(){
  final linkedList = LinkedList<int>();

  linkedList.add(10);
  linkedList.add(20);
  linkedList.add(30);

  linkedList.printList();
  linkedList.remove(20);
  linkedList.printList();
  
}